tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/DynamoDb/BatchGetItemAsyncIntegration.cs

```diff
@@ -0,0 +1,90 @@
+// <copyright file="BatchGetItemAsyncIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.ComponentModel;
+using System.Threading;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.DynamoDb
+{
+    /// <summary>
+    /// AWSSDK.DynamoDBv2 BatchGetItemAsync CallTarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "AWSSDK.DynamoDBv2",
+        TypeName = "Amazon.DynamoDBv2.AmazonDynamoDBClient",
+        MethodName = "BatchGetItemAsync",
+        ReturnTypeName = "System.Threading.Tasks.Task`1[Amazon.DynamoDBv2.Model.BatchGetItemResponse]",
+        ParameterTypeNames = new[] { "Amazon.DynamoDBv2.Model.BatchGetItemRequest", ClrNames.CancellationToken },
+        MinimumVersion = "3.0.0",
+        MaximumVersion = "3.*.*",
+        IntegrationName = AwsDynamoDbCommon.IntegrationName)]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class BatchGetItemAsyncIntegration
+    {
+        private const string Operation = "BatchGetItem";
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TBatchGetItemRequest">Type of the request object</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method</param>
+        /// <param name="request">The request for the DynamoDB operation</param>
+        /// <param name="cancellationToken">CancellationToken value</param>
+        /// <returns>CallTarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTarget, TBatchGetItemRequest>(TTarget instance, TBatchGetItemRequest request, CancellationToken cancellationToken)
+            where TBatchGetItemRequest : IBatchRequest, IDuckType
+        {
+            if (request.Instance is null)
+            {
+                return CallTargetState.GetDefault();
+            }
+
+            var scope = AwsDynamoDbCommon.CreateScope(Tracer.Instance, Operation, out AwsDynamoDbTags tags);
+            if (request.RequestItems.Count != 1)
+            {
+                return new CallTargetState(scope);
+            }
+
+            // TableName tagging only when batch is from one table.
+            var iterator = request.RequestItems.GetEnumerator();
```

Comment:
nit: Since we're doing the same operation against the same IBatchRequest interface, we could refactor this functionality into a new `internal` method in `AwsDynamoDbCommon.cs`. That would also make the logic testable and (if we were hardcore) benchmark-able

Reply:
Updated to create a method to avoid repeating code, thanks for the suggestion.

Also added unit tests for it. :shipit: